### ISSUE TITLE
Constant CPU usage

### ISSUE THREAD

**Issue by jan4843** [caddyserver/cache-handler#50]
Caddy consumes a constant non-insignificant amount of CPU when the `cache` handler is registered.

Built with:

```console
$ xcaddy build v2.6.4 --with github.com/caddyserver/cache-handler@v0.6.0
```

Caddyfile:

```
{
	order cache before rewrite
}

:8080 {
	cache
	respond "hello"
}
```

Unexpected CPU usage:

```console
$ ./caddy run &
$ top -p $!
    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
 259235 root      20   0 1776392  85420  40576 S   6.2   2.5   0:17.95 caddy
```

The CPU % is a bit exaggerated in the example because that's running on a wimpy AMD GX-424CC (Linux x86_64), but also on an Apple M2 it constantly consumes ~1% while idle (macOS arm64).

When the `cache` configuration is removed from the Caddyfile, Caddy idles at virtually 0% CPU.

---

I've profiled Caddy in the situation with Go `pprof`, but haven't investigated further: [cpu.pprof](https://github.com/caddyserver/cache-handler/files/11066075/cpu.pprof.gz).

![](https://github.com/caddyserver/cache-handler/files/11066093/profile001.pdf)

**Comment by darkweak** [comment]
Hello @jan4843, that CPU usage comes from the badger instance that is running.

**Comment by darkweak** [comment]
Yes that's the default configuration but you can override that using the badger configuration directive. I wonder if I should set nutsdb as the default storage in a future release.

**Comment by jan4843** [comment]
It's safe to say the issue is not of the cache-handler module then.

Thanks for the replies and for your software.
